app/scripts/components/breadcrumbs/breadcrumb.tsx

```diff
@@ -1,40 +1,79 @@
 import * as React from 'react';
 import './breadcrumb.scss';
-import {projectsListMoc} from '../../constants/moc';
-import {BREAD_CRUMB_SEPARATOR, COUNT_PROJECT_TITLE} from '../../constants/constants';
+import {COUNT_PROJECT_TITLE} from '../../constants/constants';
+import {useDispatch, useSelector} from 'react-redux';
+import {useEffect, useState} from 'react';
+import {clearSelectedProjectId} from '../../redux/reducers/selectedProject/selectedProject.actionCreator';
+import {clearSelectedPageId} from '../../redux/reducers/selectedPage/selectedPage.actionCreator';
 
-export const Breadcrumb = ({crumbs, selected}: { crumbs: string[]; selected: (crumb: string) => void }) => {
-  const isLastCrumb = (index: number) => index === crumbs.length - 1;
+export const Breadcrumb = () => {
+  // @ts-ignore
+  const projects = JSON.parse(localStorage.getItem('projects'));
+  // @ts-ignore
+  const pages = JSON.parse(localStorage.getItem('pages'));
+  const projectLength = projects.length;
+  const [projectName, setProjectName] = useState('Empty');
+  const [pageName, setPageName] = useState('Empty');
 
-  const getTitle = (projectsList: any []) => {
-    const listLength = projectsList.length;
-    return listLength > 1 ? `${COUNT_PROJECT_TITLE}s` : `${COUNT_PROJECT_TITLE}`;
+  // @ts-ignore
+  const {selectedPageId, selectedProjectId} = useSelector(state => state);
+  const dispatch = useDispatch();
+
+  const getTitle = () => {
+    return projectLength > 1 ? `${COUNT_PROJECT_TITLE}s` : `${COUNT_PROJECT_TITLE}`;
   };
 
-  const crumbsArray = crumbs.map((crumb, index) => {
-    return (
-      <li key={index}>
-        <button
-          className={`breadcrumbButton ${isLastCrumb(index) && 'breadcrumbLast'}`}
-          onClick={() => selected(crumb)}
-          disabled={isLastCrumb(index)}>
-          {crumb}
-        </button>
-        {!isLastCrumb(index) && <span> {' >'}</span>}
-      </li>
-    );
-  });
+  useEffect(() => {
```

Comment:
useEffect for the project and the page have the same structure, it might be better to use one function instead of two.